scipy/stats/_continuous_distns.py

```diff
@@ -1451,7 +1451,26 @@ def _shape_info(self):
 
     def _pdf(self, x):
         # cauchy.pdf(x) = 1 / (pi * (1 + x**2))
-        return 1.0/np.pi/(1.0+x*x)
+        with np.errstate(over='ignore'):
+            return 1.0/np.pi/(1.0+x*x)
+
+    def _logpdf(self, x):
+        # The formulas
+        #     log(1/(pi*(1 + x**2))) = -log(pi) - log(1 + x**2)
+        #                            = -log(pi) - log(x**2*(1 + 1/x**2))
+        #                            = -log(pi) - (2log(|x|) + log1p(1/x**2))
+        # are used here.
+        absx = np.abs(x)
+        # In the following _lazywhere, `f` provides better precision than `f2`
+        # for small and moderate x (although `f2` isn't too bad), while `f2`
+        # avoids the overflow that can occur with absx**2. An overflow of absx**2
+        # in the expression 1/absx**2 is not a problem, so we can ignore it.
+        with np.errstate(over='ignore'):
+            y = _lazywhere(absx < 1, (absx,),
+                           f=lambda absx: -_LOG_PI - np.log1p(absx**2),
+                           f2=lambda absx: (-_LOG_PI -
+                                            (2*np.log(absx) + np.log1p(1/absx**2))))
```

Comment:
The `with errstate` guard could be eliminated if `1/x**2` is replaced by (`1/x)**2`, but the current code also works without problem!

Reply:
Yup, good idea.  I included this in the recent update.